cairo/ethereum/crypto/hash.cairo

```diff
@@ -1,3 +1,20 @@
-from ethereum.base_types import Bytes32
+from ethereum.base_types import Bytes32, Bytes, Uint256
+from src.utils.bytes import bytes_to_bytes8_little_endian
+from starkware.cairo.common.builtin_keccak.keccak import keccak_bigend
+from starkware.cairo.common.cairo_builtins import BitwiseBuiltin, KeccakBuiltin
+from starkware.cairo.common.alloc import alloc
 
 using Hash32 = Bytes32;
+
+func keccak256{range_check_ptr, bitwise_ptr: BitwiseBuiltin*, keccak_ptr: KeccakBuiltin*}(
+    buffer: Bytes
+) -> Hash32 {
+    alloc_locals;
+    let (local dst: felt*) = alloc();
+    bytes_to_bytes8_little_endian(dst, buffer.value.len, buffer.value.data);
+
+    let (code_hash) = keccak_bigend(dst, buffer.value.len);
+    tempvar value = new Uint256(low=code_hash.low, high=code_hash.high);
```

Comment:
can use `&` instead of a new Uint256 ?

Reply:
not possible, even with `__fp__`. The only other option is to `get_ap()` then `cast(ap, Uint256*)`. 

`get_ap` does two function calls, while `new` only one (`alloc`), at the end i think that there is a significant different difference (given that the whole function consumes much more in any case)